### PR TITLE
add lg participants to weekly sms advert

### PR DESCRIPTION
**Pull request by IsaackMwenda**
Ideally, all the lg participants should have sent a message but that is not always the case. This allows us to advertise to all listening to group participants by adding them to the weekly SMS ad CSV file. 
 + other minor code and args refactor.

### CODE REVIEW

File: export_kakuma_advert_contacts.py [hunk 1]
```diff
@@ -72,35 +74,57 @@
...
 
+    # Load listening group de-identified CSV files
+    listening_group_csvs = []
+    for listening_group_csv_url in pipeline_configuration.listening_group_csv_urls:
+        listening_group_csvs.append(listening_group_csv_url.split("/")[-1])
+
+    for listening_group_csv in listening_group_csvs:
+        with open(f'{data_dir}/Raw Data/{listening_group_csv}', "r", encoding='utf-8-sig') as f:
+            data = list(csv.DictReader(f))
+            log.info(f'Loaded {len(data)} ' f'{data_dir}/Raw Data/{listening_group_csv} listening group participants')
```
Comment (as2388) [inline export_kakuma_advert_contacts.py]:
```suggestion
            log.info(f'Loaded {len(data)} listening group participants from {data_dir}/Raw Data/{listening_group_csv}')
```

File: export_kakuma_advert_contacts.py [hunk 2]
```diff
@@ -24,18 +24,20 @@
...
                         help="Path to WUSC-KEEP-II-KAKUMA pipeline configuration json file")
-    parser.add_argument("messages_json_input_path", metavar="messages_json_input_path",
-                        help="Path to the WUSC-KEEP-II-KAKUMA messages traced data JSONL file to extract phone"
-                             "numbers from")
-    parser.add_argument("contacts_csv_path", metavar="contacts-csv-path",
-                        help="CSV file path to write the contacts data to")
+    parser.add_argument("data_dir", metavar="data-dir",
+                        help="Directory path to read messages traced data JSONL file + listening group CSV files "
+                             "to extract phone from, and write the advert CSV file to")
+    parser.add_argument("sms_ad_flow_name", metavar="sms-ad-flow-name",
```
Comment (as2388) [inline export_kakuma_advert_contacts.py]: Why did this need to change from a path to a flow name?

File: export_kakuma_advert_contacts.py [hunk 3]
```diff
@@ -72,35 +74,57 @@
     swahili_uuids = set()
     all_uuids = set()
 
+    # Load listening group de-identified CSV files
```
Comment (as2388) [inline export_kakuma_advert_contacts.py]:
```suggestion
    # Load all listening group de-identified CSV files
```

**Review by as2388 — CHANGES_REQUESTED** [review]
What happens if someone is coded with one language in the listening group but a different one in the traced data? Will they be added to the advert csv twice (and then get the advert twice/the wrong language?)

**Comment by IsaackMwenda** [comment]
> What happens if someone is coded with one language in the listening group but a different one in the traced data? Will they be added to the advert csv twice (and then get the advert twice/the wrong language?)

Chatting with the team we have resolved to give precedence to the manually labeled language because that allows us to talk to the participant in the language that they prefer. We have seen many cases of e.g Sudanese choosing English as their preferred language etc.  Have updated the code to effect this.

File: export_kakuma_advert_contacts.py (hunk 2 above)
Comment (IsaackMwenda) [reply export_kakuma_advert_contacts.py]: I have refactored the code to read from the main data directory and branch out to the individual directories where necessary. The sms_ad_flow_name is therefore used to define the name of the advert CSV file.

Comment (as2388) [reply export_kakuma_advert_contacts.py]: Hmm this breaks with convention used elsewhere in the pipeline. The rest of the pipeline's python scripts are very flexible and generally take paths to specific files. The run scripts just provide sensible defaults, including setting the data/Outputs structure. 

Flexible python scripts with sensible defaults in the run scripts still feels like the best structure to me, but as the reason for that is mostly historic I'm happy to talk about re-engineering all the scripts if necessary.

Comment (as2388) [reply export_kakuma_advert_contacts.py]: In any case, I'm not sure the data directory is the best place to export the advert contacts to. This isn't running automatically as part of the pipeline, we need to be careful not to include phone numbers in the data archives, and TextIt contains the canonical groups anyway.

**Review by as2388 — CHANGES_REQUESTED** [review]
Requesting changes because I don't think we should risk exporting phone numbers to the same directory as the one which pipelines automatically archive. Otherwise, looks good

File: export_kakuma_advert_contacts.py [hunk 4]
```diff
@@ -154,7 +182,7 @@
         }
 
     log.warning(f"Exporting {len(advert_contacts)} contacts to {contacts_csv_path}")
-    with open(contacts_csv_path, "w") as f:
+    with open(f'{contacts_csv_path}.csv', "w") as f:
```
Comment (as2388) [inline export_kakuma_advert_contacts.py]:
```suggestion
    with open(contacts_csv_path, "w") as f:
```
For consistency with the rest of the pipeline

**Review by as2388 — APPROVED** [review]

**Pull request merged**